test/nn/conv/test_rgcn_conv.py

```diff
@@ -4,31 +4,43 @@
 import torch_geometric.typing
 from torch_geometric.nn import FastRGCNConv, RGCNConv
 from torch_geometric.testing import is_full_test
-from torch_geometric.typing import SparseTensor
-
+from torch_geometric.typing import WITH_PT2, SparseTensor
+
+DEVICES = [torch.device('cpu')]
+if torch.cuda.is_available():
+    DEVICES.append(torch.device('cuda'))
+    import os
+    os.environ['NVIDIA_TF32_OVERRIDE'] = '0'
+    torch.backends.cuda.matmul.allow_tf32 = False
+    minor_vers = str(torch.__version__).split('.')[1]
+    if WITH_PT2 or minor_vers >= 12:  # This only exists after 1.12
+        torch.set_float32_matmul_precision('highest')  # Enforce FP32
```

Comment:
Curious, what is happening here? Why is this needed?

Reply:
in nvidia containers, default matmuls act in tf32 and we were getting numerical mismatches. I have the same code in pyg-lib (thats where i copypasted from): https://github.com/pyg-team/pyg-lib/blob/master/test/ops/test_matmul.py#L7-L19